### ISSUE TITLE
Think hard about how to handle CHOICE

### ISSUE THREAD

**Issue by alex** [alex/rust-asn1#19]

**Comment by alex** [comment]
Fundamentally the only possible API here is:

- Peek at tag
- give that to callback, let it do stuff

Implementation is straightforward pending figuring out implicit/explicit tagging

**Comment by alex** [comment]
I think you can basically treat a choice as a sequence of optionals (possibyl with explicit/implicit tags) where at least one must have a value.

**Comment by alex** [comment]
Ok, better plan:

```rs
enum Choice2<T1: Asn1Element, T2: Asn1Element> {
    Element1(T1::Output),
    Element2(T2::Output),
}
```

we can generate 2-5 of these or so.

**Comment by alex** [comment]
This worked out easily enough, solved in #54
